numpy provides lots of mathematical functions which are vectorized, just like the arithmetic operations, which means they operate elementwise. They return new arrays of the same size and shape with elements of the same type or another type if necessary.

There are really lots of mathematical functions, let's just have a look at a narrow selection of them. Others will be used in the future when need arises.

Let's start by creating a simple vector array on which we can demonstrate how the functions work.

In [2]:
import numpy as np

# Here's a simple array.
A = np.array([1, 2, 10, 20])
A

array([ 1,  2, 10, 20])

In [3]:
# And now let's demonstrate some functions. 
# First np.sqrt, which returns the square root.
np.sqrt(A)

array([1.        , 1.41421356, 3.16227766, 4.47213595])

In [5]:
# Next np.exp, which returns the exponential.
np.exp(A)

array([2.71828183e+00, 7.38905610e+00, 2.20264658e+04, 4.85165195e+08])

In [6]:
# Then we have the logarithms in a couple of flavors, like the natural logarithm (np.log), 
# the logarithm in base 2 (np.log2) or the logarithm in base 10 (np.log10):
np.log2(A)

array([0.        , 1.        , 3.32192809, 4.32192809])

In [11]:
# Then we have the trigonometric functions, as well as the inverse, hyperbolic and inverse hyperbolic trigonometric functions.
# Here's an example with the sine function - here the values are in radians.
np.sin(A)

array([ 0.84147098,  0.90929743, -0.54402111,  0.91294525])

In [12]:
# The next function is np.round, which is used to round numbers to a given number of decimal places. Let's use it
# along with the result of the sine function.
np.round(np.sin(A), decimals = 2)

array([ 0.84,  0.91, -0.54,  0.91])

EXERCISE

Here's a simple array:

X = np.linspace(1, np.e, 10)

where np.e is a constant, the Euler's number, approximately equal to 2.71828.

Using some of the mathematical functions presented above create an array of natural logarithms of the numbers from the X array, rounded to 3 decimal places.

SOLUTION

In [19]:
X = np.linspace(1, np.e, 10)

np.round(np.log(X), decimals = 3)

array([0.   , 0.175, 0.323, 0.453, 0.567, 0.67 , 0.763, 0.849, 0.927,
       1.   ])